### PR TITLE
Add postprocessing of lumped port boundaries to eigenmode simulations

### LINKED ISSUE #123: Output lumped port voltages and currents for eigenmode simulations
These are computed and output as functions of frequency in `port-V.csv` and `port-I.csv` for  frequency domain simulations. The projections of eigenmodes onto these variables is useful for reconstructing things like scattering matrices from eigenmode results.

### PR DESCRIPTION
**Pull request by sebastiangrimberg**
These quantities can be useful in addition to the EPR output for characterizing multi-port systems along with the computed pole locations.

This PR also updates the output of various computed quantities like energies, voltages, currents, and field values to be dimensional based on a choice for `H0`, the characteristic magnetic field strength used for nondimensionalization (see https://awslabs.github.io/palace/stable/reference/). In particular, we choose `H0` so that the unit power incident on ports for driven simulations corresponds to 1 W. The fields written for ParaView visualization are *NOT* redimensionalized, however. They can be if this would be useful, but for now we leave them nondimensional since they are only used for visualization.

Resolves #123.

### CODE REVIEW

File: palace/drivers/basesolver.cpp
Comment (sebastiangrimberg) [reply palace/drivers/basesolver.cpp:435]: Yep, should be V/m for E. Was distracted doing units of B instead of H and so wrote units for D instead of E. Good catch.

File: palace/utils/iodata.cpp
Comment (sebastiangrimberg) [reply palace/utils/iodata.cpp:612]: I don't mind the break's enough to warrant the extra syntax. Many ways to accomplish the same thing and I'll leave as is, thanks for your comment.

File: docs/src/guide/problem.md
Comment (sebastiangrimberg) [reply docs/src/guide/problem.md:80]: This change would not be correct and would change the meaning of the sentence. This is referring to time domain port processing.

File: docs/src/guide/postprocessing.md
Comment (sebastiangrimberg) [reply docs/src/guide/postprocessing.md:43]: I think it's fine as is, separated to emphasize ports vs. surface currents.

File: palace/drivers/basesolver.cpp
Comment (sebastiangrimberg) [reply palace/drivers/basesolver.cpp:436]: Sure, could do that too. Doesn't matter either way and will leave as is.

File: palace/utils/iodata.cpp
Comment (sebastiangrimberg) [reply palace/utils/iodata.cpp:559]: `std::sqrt` is not always `constexpr` and you don't gain much (anything) here from this being `constexpr`. Fine as is.

File: palace/utils/iodata.hpp
Comment (sebastiangrimberg) [reply palace/utils/iodata.hpp:78]: Commenting here along with https://github.com/awslabs/palace/pull/129#discussion_r1392851527, but yes originally I had templated for complex but then decided on this design being nicer. I don't see any penalty for it being templated, so will keep it. If someone is converting an integral type they can add an explicit instantiation when they come upon the linker error.

File: docs/src/guide/problem.md
Comment (hughcars) [reply docs/src/guide/problem.md:80]: Then it would be "As with"? As for is describing the frequency domain, as with is drawing similarity to the frequency domain case.

File: palace/utils/iodata.hpp
Comment (hughcars) [reply palace/utils/iodata.hpp:78]: The function doesn't however work for other types because all the constants in `electromagnetics` are double. Adding an explicit instantiation for `int` for instance gives
```
  const auto f1 = iodata.DimensionalizeValue(IoData::ValueType::FREQUENCY, 1.0);
  const auto f2 = iodata.DimensionalizeValue(IoData::ValueType::FREQUENCY, 1);
  std::cout << "f1 " << f1 << " f2 " << f2 << std::endl;
```
prints `f1 0.870683 f2 0` due to the rules on integer multiplication/division and casting. Dropping the template for `double` performs the conversions and gives `f1 0.870683 f2 0.870683` as expected. The extra template syntax buys unexpected behaviour, or requires a user to exercise extra care on numeric types, neither seems ideal.

Comment (sebastiangrimberg) [reply palace/utils/iodata.hpp:78]: OK maybe it was a bad quick example, you'd never want to instantiate this function for an `int` and if you did then the output you are getting is actually exactly what you'd expect and I argue is actually correct behavior (the returned type should be integral, not floating point in this case). I have a hard time seeing where developers will not be able to take the care to work with the function as it currently exists.

**Comment by sebastiangrimberg** [comment]
@hughcars, any final comments here that need to be addressed before approval?

**Review by hughcars — APPROVED** [review]

**Pull request merged**
